### PR TITLE
[RFC] add inplace functions

### PR DESCRIPTION
**Pull request by mkborregaard**
As suggested on the PR - this is made towards #1 but implements in-place versions of landscape
(sorry Tim, no emojis)

### CODE REVIEW

File: src/landscape.jl [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+rescale!(mat) = (mat .- minimum(mat)) ./ (maximum(mat) - minimum(mat))
```
Comment (mkborregaard) [inline src/landscape.jl]: I still think we might use the StatsBase implementation instead

File: src/landscape.jl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+function mask!(array, maskarray) 
+    for i in eachindex(array, maskarray)
+        array[i] = maskarray[i] == 0 ? array[i] : missing
```
Comment (mkborregaard) [inline src/landscape.jl]: @tpoisot what do you think here? this implementation causes problems if the type of `array` does not include `missing`. But restricting to `NaN` requires us to have `Float64`.  How to make this general?

Comment (tpoisot) [reply src/landscape.jl]: This is the biggest problem with `SimpleSDMLayers` at the moment as well (we use `nothing` internally). I think it is safe to assume that the landscapes here are going to floating points, so using `NaN` (or `NaN32` etc) is not too bad in this specific case. The other solution is to go full sparse, but this assuming the masked part is going to be small it's almost never going to be worth it.

Comment (tpoisot) [reply src/landscape.jl]: In fact I'd go even further and assume that the mask should be a matrix of `AbstractBool`, and not something where we need to look at the `zero` of the type.

Comment (tpoisot) [reply src/landscape.jl]:
~~~julia
function mask!(array, maskarray)
    array[findall(.!maskarray)] .= NaN
end
~~~

?

File: src/landscape.jl (hunk 1 above)
Comment (tpoisot) [reply src/landscape.jl]: absolutely, yes

File: src/landscape.jl (hunk 2 above)
Comment (mkborregaard) [reply src/landscape.jl]: no that allocates a temporary array

Comment (mkborregaard) [reply src/landscape.jl]: but without the `findall` I guess it should work?

**Comment by mkborregaard** [comment]
StatsBase rescaling doesn't work for the whole matrix. So I've just made the very minor edit of inserting NaN instead and assuming the mask is boolean. What do you say we add some tests and then merge #1 and #2 at this point?

File: src/landscape.jl (hunk 1 above)
Comment (mkborregaard) [reply src/landscape.jl]: I just asked on the StatsBase PR that added standardization and they suggest using the simple broadcasting here https://github.com/JuliaStats/StatsBase.jl/pull/85#issuecomment-777213575

**Pull request merged**